specs/WebNotification.md

```diff
@@ -0,0 +1,610 @@
+  [propget] HRESULT Notification([out, retval] ICoreWebView2Notification** notification);
+  
+  /// Sets whether the `NotificationReceived` event is handled by the host after
+  /// the event handler completes or if there is a deferral then after the
+  /// deferral is completed.
+  ///
+  /// If `Handled` is set to TRUE then WebView will not display the notification
+  /// with the default UI, and the host will be responsible for handling the
+  /// notification and let the web content know that the notification has been
+  /// displayed, clicked, or closed. If after the event handler or deferral
+  /// completes `Handled` is set to FALSE then WebView will display the default
+  /// notification. Note that if `Show` has been called on the `Notification`
+  /// object, WebView will not display the default notification regardless of
+  /// the Handled property. The default value is FALSE.
+  [propput] HRESULT Handled([in] BOOL value);
+
+  /// Gets whether the `NotificationReceived` event is handled by host.
+  [propget] HRESULT Handled([out, retval] BOOL* value);
+
+  /// Returns an `ICoreWebView2Deferral` object. Use this operation to complete
+  /// the event at a later time.
+  HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+}
+
+/// An event handler for the `Closed` event.
+[uuid(6A0B4DE9-8CBE-4211-BAEF-AC037B1A72DE), object, pointer_default(unique)]
+interface ICoreWebView2NotificationClosedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2Notification* sender,
+      [in] IUnknown* args);
+}
+
+/// A collection of notification actions.
+[uuid(89D8907E-18C7-458B-A970-F91F645E4C43), object, pointer_default(unique)]
+interface ICoreWebView2NotificationActionCollection : IUnknown {
+  /// The number of notification actions contained in the
+  /// ICoreWebView2NotificationActionCollection.
+  [propget] HRESULT Count([out, retval] UINT* count);
+
+  /// Gets the notification action at the given index.
+  HRESULT GetValueAtIndex([in] UINT index,
+                          [out, retval] COREWEBVIEW2_NOTIFICATION_ACTION* value);
+}
+
+/// A collection of unsigned long integers.
+[uuid(974A91F8-A309-4376-BC70-7537D686533B), object, pointer_default(unique)]
+interface ICoreWebView2UnsignedLongCollection : IUnknown {
+  /// The number of unsigned long integers contained in the
+  /// ICoreWebView2UnsignedLongCollection.
+  [propget] HRESULT Count([out, retval] UINT* count);
+
+  /// Gets the unsigned long integer at the given index.
+  HRESULT GetValueAtIndex([in] UINT index, [out, retval] UINT64* value);
+}
+
+/// This is the ICoreWebView2Notification that represents a [HTML Notification
+/// object](https://developer.mozilla.org/docs/Web/API/Notification).
+[uuid(E3F43572-2930-42EB-BD90-CAC16DE6D942), object, pointer_default(unique)]
+interface ICoreWebView2Notification : IUnknown {
+  /// Add an event handler for the `Closed` event.
```

Comment:
What does it mean that the web code closed the notification? Is it just that they called a close method and there's no side effects? 

For the case of the host app doing custom notification UI this seems like this event should be better called `CloseRequested`. That is, the web code wants the notification to be closed and the host app needs to handle that request and remove the notification from the screen (or whatever)

Does this also apply in the case of default notification UI? What does the Closed event mean in that case and what is it useful for in that case?

Reply:
I believe `CloseRequested` would make sense here. "... the web code closed the notification" means the notification object is closed through Notification instance method `close()`, such as `event.notification.close()`. In the case of default notification UI, the browser logic will close the notification accordingly.